src/main/java/seedu/contax/logic/commands/RangeCommand.java

```diff
@@ -67,13 +69,18 @@ public CommandResult execute(Model model) throws CommandException {
             AddressBookParser addressBookParser = new AddressBookParser();
             try {
                 String commandText = ParserUtil.parseAndCreateNewCommand(commandInput, Integer.toString(i));
-
                 logger.info("----------------[RANGE COMMAND][" + commandText + "]");
-
                 Command command = addressBookParser.parseCommand(commandText);
-                commandResultList.add(command.execute(model));
+                try {
+                    commandResultList.add(command.execute(model));
+                } catch (CommandException ce) {
+                    commandResultList.clear(); // only for the purpose of not changing output in feature freeze
+                    setFirstOccurrencePerson(model, restorePerson);
+                    commandResultList.add(new CommandResult(ce.getMessage()));
+                    break;
+                }
             } catch (ParseException pe) {
-                return new CommandResult(pe.getMessage());
+                throw new CommandException(pe.getMessage());
```

Comment:
Does it save if you do this?

Reply:
parse error shoud not process at first place

Reply:
Fair point.